### DISCUSSION TITLE
Script execution delay in the n9e exec module

### DISCUSSION THREAD

**Discussion by wushengwusheng** [ccfos/nightingale#1358] [Q&A] [answered]
一个脚本设定3分钟执行一次，每次执行耗费20s，会导致其他频率的脚本同步顺延20s，这样的话就会相应导致延迟告警，请问有什么办法处理吗？
<img width="177" alt="image" src="https://user-images.githubusercontent.com/30250875/212583340-0f4b5641-2e94-48b8-b6c0-7a0102e4d3d0.png">
<img width="656" alt="image" src="https://user-images.githubusercontent.com/30250875/212583545-3398ec1e-bce6-45a7-8f09-9ca166b607b0.png">

**Comment by wushengwusheng** [comment]
目前就是就是有这种模式吗？所有脚本按所设置的频次执行，在设定timeout内执行完，不同频次脚本之间互不影响，有这种配置吗？

**Comment by UlricQin** [comment, marked as answer]
目前只有一种方式，就是顺延的

**Reply by wushengwusheng** [reply]
后续会有计划优化这个方向吗？顺延的话，会不会造成告警延迟或者图断点之类的问题？
